conda/common/configuration.py

```diff
@@ -1609,3 +1609,76 @@ def typify_parameter(self, parameter_name, value, source):
 
     def get_descriptions(self):
         raise NotImplementedError()
+
+
+def get_plugin_config_data(
+    data: dict[Path, dict[str, RawParameter]],
+) -> dict[Path, dict[str, RawParameter]]:
+    """
+    This is used to move everything under the key "plugins" from the provided dictionary
+    to the top level of the returned dictionary. The returned dictionary is then passed
+    to :class:`PluginConfig`.
+    """
+    new_data = defaultdict(dict)
+
+    for source, config in data.items():
+        if plugin_data := config.get("plugins"):
+            plugin_data_value = plugin_data.value(None)
+
+            if not isinstance(plugin_data_value, frozendict):
+                continue
+
+            for param_name, raw_param in plugin_data_value.items():
+                new_data[source][param_name] = raw_param
+
+        if source == EnvRawParameter.source:
```

Comment:
We would never expect both `config.get("plugins")` and `source == EnvRawParameter.source` to be true for the same source (right?), so we should combine these for clarity:

```
        if plugin_data := config.get("plugins"):
            ...
        elif source == EnvRawParameter.source:
            ...
```